permissions_validators/src/private_blockchain/query.rs

```diff
@@ -1,371 +1,306 @@
 //! Query Permissions.
 
+use iroha_core::smartcontracts::permissions::ValidatorVerdict;
+
 use super::*;
 
 /// Allow queries that only access the data of the domain of the signer.
 #[derive(Debug, Copy, Clone, Serialize)]
 pub struct OnlyAccountsDomain;
 
-impl IsAllowed<QueryBox> for OnlyAccountsDomain {
-    #[allow(clippy::too_many_lines, clippy::match_same_arms)]
+impl IsAllowed for OnlyAccountsDomain {
+    type Operation = QueryBox;
+
+    #[allow(
+        clippy::too_many_lines,
+        clippy::match_same_arms,
+        clippy::cognitive_complexity
+    )]
     fn check(
         &self,
         authority: &AccountId,
         query: &QueryBox,
         wsv: &WorldStateView,
-    ) -> Result<(), DenialReason> {
+    ) -> ValidatorVerdict {
         use QueryBox::*;
-        let context = Context::new();
         match query {
             FindAssetsByAssetDefinitionId(_) | FindAssetsByName(_) | FindAllAssets(_) => {
-                Err("Only access to the assets of the same domain is permitted."
-                    .to_owned()
-                    .into())
-            }
-            FindAllAccounts(_) | FindAccountsByName(_) | FindAccountsWithAsset(_) => Err(
-                "Only access to the accounts of the same domain is permitted."
-                    .to_owned()
-                    .into(),
-            ),
-            FindAllAssetsDefinitions(_) => Err(
-                "Only access to the asset definitions of the same domain is permitted."
-                    .to_owned()
-                    .into(),
+                Deny("Only the access to the assets of the same domain is permitted.".to_owned())
+            }
+            FindAllAccounts(_) | FindAccountsByName(_) | FindAccountsWithAsset(_) => {
+                Deny("Only the access to the accounts of the same domain is permitted.".to_owned())
+            }
+            FindAllAssetsDefinitions(_) => Deny(
+                "Only the access to the asset definitions of the same domain is permitted."
+                    .to_owned(),
             ),
-            FindAllDomains(_) => Err("Only access to the domain of the account is permitted."
-                .to_owned()
-                .into()),
-            FindAllRoles(_) => Err("Only access to roles of the same domain is permitted."
-                .to_owned()
-                .into()),
-            FindAllRoleIds(_) => Ok(()), // In case you need to debug the permissions.
-            FindRoleByRoleId(_) => Err("Only access to roles of the same domain is permitted."
-                .to_owned()
-                .into()),
-            FindAllPeers(_) => Ok(()), // Can be obtained in other ways, so why hide it.
-            FindAllActiveTriggerIds(_) => Ok(()),
+            FindAllDomains(_) => {
+                Deny("Only the access to the domain of the account is permitted.".to_owned())
```

Comment:
```suggestion
                Deny("Only access to the domain of the account is permitted.".to_owned())
```

Similarly for other error messages.